lib/ansible/modules/clustering/kubevirt/kubevirt_vm.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# Apache License, Version 2.0
+# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: kubevirt_vm
+
+short_description: Manage KubeVirt virtual machine
+
+description:
+    - Use Openshift Python SDK to manage the state of KubeVirt virtual machines.
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+options:
+    state:
+        description:
+            - Set the virtual machine to either C(present), C(absent), C(running) or C(stopped).
+        default: "present"
+        choices:
+            - present
+            - absent
+            - running
+            - stopped
+        type: str
+    name:
+        description:
+            - Name of the virtual machine.
+        required: true
+        type: str
+    namespace:
+        description:
+            - Namespace where the virtual machine exists.
+        required: true
+        type: str
+    disks:
+        description:
+            - List of dictionaries which specify disks of the virtual machine.
+            - A disk can be made accessible via four different types: I(disk), I(lun), I(cdrom), I(floppy).
+            - https://kubevirt.io/api-reference/master/definitions.html#_v1_disk
+        type: list
+    volumes:
+        description:
+            - Volume specification for the virtual machine.
+            - https://kubevirt.io/api-reference/master/definitions.html#_v1_volume
+        type: str
+    persistent:
```

Comment:
Why do we need this? How do we map that to vm/vmi definition?

Reply:
In case user do following:

```yaml
kubevirt_vm:
  name: myvm
  state: running
  memory: 64M
   disks:
     - ...
```

With `persistent=true/false` he can decide if he want to create VM object and also VMI, or he want to create only VMI object without VM object.

Reply:
The case you didn't understand from documentation means, it's wrong documentation, I will try to improve.

Reply:
I think user can decide by creating specific object VM or VMI. Not sure about the value of this flag.

Reply:
Yes, but where do he specify it?

Reply:
Actually I think we should hide this kind of implementation detail from him, so it's more user friendly to work with the module for him.

Reply:
You are right it is better to hide this implementation detail. I would consult the name with UX and let's keep it.

Reply:
1. The arg description is missing info that this defaults to True.
2. I'm not sure 'persistent' is the best term for this, though I can't really think of a better one. (Ephemeral? But that's usually used for storage, not vms.)
3. The description should contain short wording describing what implementation detail is being hidden here, so that people who actually understand how kubevirt operates don't have to guess what this does. Something like: _Persistent VMs create a VirtualMachine object in the cluster and a matching VirtualMachineInstance whenever state is /running/. Non–persistent VMs create only the latter object, which gets irrecoverably deleted whenever state is set to anything other than /running/._

Reply:
Seems like `ephemeral` is good name for this. I would update the documentation stating it is `False` be default.

Reply:
yeah `ephemeral` is used on many places in kubevirt doc, so would be the best to use it as well